### PR TITLE
Add support for adding and removing identities from SSH agents.

### PR DESCRIPTION
**Pull request by dtbartle**

### CODE REVIEW

File: lib/net/ssh/authentication/agent.rb [hunk 1]
```diff
@@ -139,6 +147,37 @@ def sign(key, data)
       return reply.read_string
     end
 
+    # Adds the private key with comment to the agent.
+    # If opts[:lifetime] is given, the key will automatically be removed after
+    # opts[:lifetime] seconds.
+    # If opts[:confirm] is true, confirmation will be required for each agent
+    # signing operation.
+    def add_identity(priv_key, comment, opts={})
```
Comment (mfazekas) [inline lib/net/ssh/authentication/agent.rb]: instead of opt, we could use kw arguments? what do you think? We now require ruby 2.0.

Comment (dtbartle) [inline lib/net/ssh/authentication/agent.rb]: Sounds good; updated the PR.

**Comment by mfazekas** [comment]
Thanks for the quality PR again!

**Comment by dtbartle** [comment]
@mfazekas I noticed https://github.com/net-ssh/net-ssh/pull/487/commits/09abd0afea40fda90262311825167754fa9e0ecf didn't get merged; should I send a separate PR for that?

**Pull request merged**